web/src/postprocess_content.ts

```diff
@@ -53,12 +53,25 @@ export function postprocess_content(html: string): string {
         }
 
         if (elt.querySelector("img") || elt.querySelector("video")) {
+            // Rewrite the legacy .message_inline_image class, whose name would add
+            // confusion when Zulip supports inline images via standard Markdown.
+            // We further adjust this class below for when the element contains a
+            // video.
+            elt.parentElement?.classList.replace(
+                "message_inline_image",
+                "message-media-preview-image",
+            );
             // We want a class to refer to media links
             elt.classList.add("media-anchor-element");
             // Add a class to the video, if it exists, including
             // the .media-image-element class for properly treating
             // video thumbnails
             if (elt.querySelector("video")) {
+                // We use a different class name to distinguish videos from images
+                elt.parentElement?.classList.replace(
+                    "message-media-preview-image",
+                    "message-media-preview-video",
+                );
```

Comment:
This would almost certainly read better with an if/else construction for what to replace `message_inline_image` with.